src/Spec-Core/SpecInterpreter.class.st

```diff
@@ -5,77 +5,75 @@ Class {
 	#name : #SpecInterpreter,
 	#superclass : #Object,
 	#instVars : [
-		'model',
 		'spec',
 		'arrayToInterpret',
-		'index'
-	],
-	#classInstVars : [
-		'bindings'
+		'index',
+		'presenter'
 	],
 	#category : #'Spec-Core-Utilities'
 }
 
 { #category : #accessing }
 SpecInterpreter class >> bindings [
-
-	^ self defaultBindings 
+	^ (SpecBindings value ifNil: [ self environment at: #MorphicAdapterBindings ]) new
 ]
 
 { #category : #accessing }
 SpecInterpreter class >> bindings: anAdapterBinding [
-
-	bindings := anAdapterBinding
+	self deprecated: 'Never worked.'
 ]
 
 { #category : #private }
 SpecInterpreter class >> defaultBindings [
-	
-	^ (CurrentSpecDefaultBindings value ifNil: [ self environment at: #MorphicAdapterBindings]) new
-	
+	self deprecated: 'Use #bindings instead.' transformWith: '`@receiver defaultBindings' -> '`@receiver bindings'.
+	^ self bindings
 ]
 
 { #category : #accessing }
 SpecInterpreter class >> hardResetBindings [
-
-	bindings := nil
+	self deprecated: 'Never worked'
 ]
 
 { #category : #protocol }
-SpecInterpreter class >> interpretASpec: aSpec model: aModel [
-	
-	^ self
-		interpretASpec: aSpec
-		model: aModel
-		selector: nil
+SpecInterpreter class >> interpretASpec: aSpec model: aPresenter [
+	self
+		deprecated: 'Use `interpretASpec: aSpec presenter: aPresenter` instead'
+		transformWith:
+			'`@receiver interpretASpec: `@statements1 model: `@statements2'
+				-> '`@receiver interpretASpec: `@statements1 presenter: `@statements2'.
+	^ self interpretASpec: aSpec presenter: aPresenter selector: nil
 ]
 
 { #category : #protocol }
-SpecInterpreter class >> interpretASpec: aPresenter model: aModel selector: aSelector [
-	| result |
-	
-	 [ result :=  self 
-				private_interpretASpec: aPresenter
-				model: aModel
-				selector: aSelector ] ensure: [ self hardResetBindings ].
-	^ result
+SpecInterpreter class >> interpretASpec: aSpec model: aPresenter selector: aSelector [
+	self
+		deprecated: 'Use `interpretASpec: aSpec presenter: aPresenter selector: aSelector` instead'
+		transformWith: '`@receiver interpretASpec: `@statements1 model: `@statements2 selector: `@statements3' -> '`@receiver interpretASpec: `@statements1 presenter: `@statements2 selector: `@statements3'.
+		
+	^ self interpretASpec: aSpec presenter: aPresenter selector: aSelector
 ]
 
-{ #category : #private }
-SpecInterpreter class >> private_buildWidgetFor: aComposablePresenter withSpec: aSymbol [
-	
-	^ self private_interpretASpec: (aComposablePresenter retrieveSpec: aSymbol) model: aComposablePresenter selector: aSymbol
+{ #category : #protocol }
```

Comment:
What is this protocol named `protocol` ?

Reply:
It was already like that and I did not yet cleaned the protocols